server/uinitialize.pas

```diff
@@ -587,7 +588,11 @@ procedure Initialize(Rpc: TRpcPeer; Request: TRpcRequest);
             else if (Key = 'FPCTARGET') and Reader.Str(s) then
               Options.TargetOS := s
             else if (Key = 'FPCTARGETCPU') and Reader.Str(s) then
-              Options.TargetProcessor := s;
+              Options.TargetProcessor := s
+            else if (Key = 'syntaxErrorCausesLspError') and Reader.Bool(b) then
```

Comment:
Perhaps it would be preferable to just add a single option that can take on multiple values (e.g. an int or a string), rather than adding two bools. That way it can also easily be extended without adding even more options.

E.g.
```pascal
type
  TSyntaxErrorReportingMode = (
    sermFakeCompletionItem = 0,
    sermShowMessage = 1,
    sermErrorResponse = 2
  );

var
  syntaxErrorReportingMode: TSyntaxErrorReportingMode = sermFakeCompletionItem;
```
...
```pascal
else if (Key = 'syntaxErrorReportingMode') and Reader.Number(n) then
  syntaxErrorReportingMode := TSyntaxErrorReportingMode(n);
```

Reply:
I made it split into 2 booleans, because there are 2 independent decisions, with 4 possible combinations:

1. what to answer (for the completion request)? Since we have to answer something.
2. should we do `window/showMessage`?

So the enum with 3 values that you propose would miss an option: "return fake completion and also make a window/showMessage", which is actually the default after this PR. 

Hm, though I see the advantage of simpler interface for it for users. Users probably don't care about LSP details discussed here ("window/showMessage decision is independent from what to answer"), users just care "how can I get the error message". In which case just 3 options

```
    sermFakeCompletionItem = 0,
    sermShowMessage = 1,
    sermErrorResponse = 2
```

seem simple enough. What do you imagine `sermShowMessage` should do with the response? Just answer with 0 completions?

Reply:
Ah, I see now. But I think it's not really useful to return an error if we already show a message or display a fake completion item. That may just result in the user getting the error message twice. As for what to return instead: Your suggestion of returning an empty list seems the most reasonable to me. I can't think of anything better.

Reply:
Agreed, and done :)